File: src/main/java/org/proofit/premiumcalc/calculation/TheftPremium.java
package org.proofit.premiumcalc.calculation;

import java.math.BigDecimal;

public class TheftPremium implements Premium {
    private static final BigDecimal DEFAULT_COEFFICIENT = new BigDecimal("0.11");
    private static final BigDecimal ALTERNATIVE_COEFFICIENT = new BigDecimal("0.05");
    private static final BigDecimal DEFAULT_RISK_LIMIT = new BigDecimal("15.00");

    @Override
    public BigDecimal getCoefficient(BigDecimal sumInsured) {
        return shouldDefaultRiskBeUsedFor(sumInsured) ? DEFAULT_COEFFICIENT : ALTERNATIVE_COEFFICIENT;
    }

    private static boolean shouldDefaultRiskBeUsedFor(BigDecimal sumInsuredFire) {
        return DEFAULT_RISK_LIMIT.compareTo(sumInsuredFire) > 0;
    }
}

File: src/main/kotlin/org/proofit/premiumcalc/calculation/TheftPremium.kt
package org.proofit.premiumcalc.calculation

import java.math.BigDecimal

class TheftPremium : Premium {
    private val DEFAULT_COEFFICIENT = BigDecimal("0.11")
    private val ALTERNATIVE_COEFFICIENT = BigDecimal("0.05")
    private val DEFAULT_RISK_LIMIT = BigDecimal("15.00")

    private fun isDefaultRisk(sumInsuredFire: BigDecimal): Boolean {
        return sumInsuredFire < DEFAULT_RISK_LIMIT
    }

    override fun getCoefficient(sumInsured: BigDecimal): BigDecimal {
        return if (isDefaultRisk(sumInsured)) DEFAULT_COEFFICIENT else ALTERNATIVE_COEFFICIENT
    }
}
